ygen/enumgen_test.go

```diff
@@ -751,26 +1147,33 @@ func TestResolveNameClashSet(t *testing.T) {
 				},
 			},
 		},
-		wantErrSubstr: `enumKey "enum-a" has been given a second name`,
+		inShortenEnumLeafNames: true,
+		wantErrSubstr:          `enumKey "enum-a" has been given a second name`,
 	}}
 
 	for _, tt := range tests {
 		for noUnderscores, wantUniqueNamesMap := range map[bool]map[string]string{
 			false: tt.wantUniqueNamesMap,
 			true:  tt.wantUniqueNamesMapNoUnderscores} {
 
-			inDefinedEnums := tt.inDefinedEnums
-			if noUnderscores {
-				inDefinedEnums = tt.inDefinedEnumsNoUnderscores
-			}
 			for compressPaths := range map[bool]struct{}{false: {}, true: {}} {
-				t.Run(tt.name+fmt.Sprintf("@compressPaths:%v,noUnderscores:%v", compressPaths, noUnderscores), func(t *testing.T) {
+				t.Run(tt.name+fmt.Sprintf("@compressPaths:%v,noUnderscores:%v,shortenEnumLeafNames:%v", compressPaths, noUnderscores, tt.inShortenEnumLeafNames), func(t *testing.T) {
 					s := newEnumGenState()
-					for k, v := range inDefinedEnums {
+					for k, v := range tt.inDefinedEnums {
```

Comment:
Does this logic always hold for these tests -- it seems like it probably does today, but it also looks slightly fragile insofar that it seems like there might be cases where we could end up with deduplication occurring in the proto case where we have no underscores, and uppercase being used. Am I misparsing here?

Reply:
No your observation is valid, this mapping is definitely not sound currently due to the case you pointed out, and others as well.

I didn't write this to assert this case -- my only purpose was to avoid repeating `tt.inDefinedEnumsNoUnderscores` for every test case, because none of the tests trigger these corner cases.

I see this optimization is too complex because it can cause misunderstanding due to the intrinsic complexity of `definedEnums`, so I've added `tt.inDefinedEnumsNoUnderscores` now.

Reply:
Great, thanks. I get that this is more verbose, but the general rule that I think we try to follow is that we don't introduce logic in a test wherever possible. Of course, there's some cases where that is unavoidable, or the cost of not having the logic is so high that we want to avoid it - but this looks to be a case where the duplication is verbose, but useful.